### ISSUE TITLE
Sync gradients every N iterations in DistributedDataParallel

### ISSUE THREAD

**Issue by llidev** [pytorch/pytorch#14462]
Hi,

Since my model is big, I am trying to perform all-reduce on gradient every N steps. I am not sure if DistributedDataParallel module supports this natively. 

I tried the below code, which looks like working fine... 

```python
model = torch.nn.parallel.DistributedDataParallel(model,...)
gradient_accumulation_steps = 4
for step, batch in enumerate(tqdm(train_dataloader, desc="Iteration")):
    if (step + 1) % gradient_accumulation_steps == 0:
        loss = model(batch)
    else:
        loss = model.module(batch)
    loss = loss / gradient_accumulation_steps
    loss.backward()
    tr_loss += loss.item()
    if (step + 1) % gradient_accumulation_steps == 0:
        optimizer.step()
        model.zero_grad()
```
But I'd like to checkout if you can point if there is anything wrong or provide a better implementation.

Thank you!

**Comment by ngimel** [comment]
Closing due to age
